### ISSUE TITLE
getDataSets does not include options

### ISSUE THREAD

**Issue by aguarino77** [jtblin/angular-chart.js#370]
getDataSets is not including options in the dataset object:

```
    function getDataSets (labels, data, series, colors, yaxis) {
      return {
        labels: labels,
        datasets: data.map(function (item, i) {
          var dataset = angular.extend({}, colors[i], {
            label: series[i],
            data: item
          });
          if (yaxis) {
            dataset.yAxisID = yaxis[i];
          }
          return dataset;
        })
      };
    }
```

With the current implementation no members other than label and data will be propagated, which makes impossible to set chart properties like "fill: false". 

In ChartJs 2.0 this is (line 2626)
`fill: line.custom && line.custom.fill ? line.custom.fill : (this.getDataset().fill !== undefined ? this.getDataset().fill : this.chart.options.elements.line.fill),
`

**Comment by aguarino77** [comment]
Is actually related to #309. But the proposed fix is a global settings. I would like to have a per chart/series options like initially proposed by @courchef

**Comment by jtblin** [comment]
Just use chart-options. Make sure you use the 1.0.0-alpha... series.

**Comment by aguarino77** [comment]
passing this value to chart-options does not work. It used to work in the 0.x series
`$scope.chartOptions = {datasetFill: false, animation: false, pointDot : true};`

**Comment by jtblin** [comment]
Probably because the options have changed in Chart.js 2.0 e.g. http://www.chartjs.org/docs/#line-chart-chart-options

**Comment by aguarino77** [comment]
If you read the chartjs docs you mention you see how the dataset object should contain the options 
```
var data = {
    labels: ["January", "February", "March", "April", "May", "June", "July"],
    datasets: [
        {
            label: "My First dataset",
            fill: false,
            lineTension: 0.1,
            backgroundColor: "rgba(75,192,192,0.4)",
            borderColor: "rgba(75,192,192,1)",
            borderCapStyle: 'butt',
            borderDash: [],
            borderDashOffset: 0.0,
            borderJoinStyle: 'miter',
            pointBorderColor: "rgba(75,192,192,1)",
            pointBackgroundColor: "#fff",
            pointBorderWidth: 1,
            pointHoverRadius: 5,
            pointHoverBackgroundColor: "rgba(75,192,192,1)",
            pointHoverBorderColor: "rgba(220,220,220,1)",
            pointHoverBorderWidth: 2,
            pointRadius: 1,
            pointHitRadius: 10,
            data: [65, 59, 80, 81, 56, 55, 40],
        }
    ]
};
```

But if getDataSets is defined like this:
```   
 function getDataSets (labels, data, series, colors, yaxis) {
      return {
        labels: labels,
        datasets: data.map(function (item, i) {
          var dataset = angular.extend({}, colors[i], {
            label: series[i],
            data: item
          });
          if (yaxis) {
            dataset.yAxisID = yaxis[i];
          }
          return dataset;
        })
      };
    }
```

I see no way to pass an option like "fill: false".
The function above should contain something like:
```
 function getDataSets (labels, data, series, colors, yaxis, options ) {
      return {
        labels: labels,
        datasets: data.map(function (item, i) {
          var dataset = angular.extend({}, colors[i], {
            label: series[i],
            data: item
          }, options);
          if (yaxis) {
            dataset.yAxisID = yaxis[i];
          }
          return dataset;
        })
      };
    }

**Comment by jtblin** [comment]
All options can be changed globally or individually. For a line, it looks like it would be `elements.line.fill`. See for an example of how to set options for a specific chart: https://github.com/jtblin/angular-chart.js/blob/chartjs-2.0/examples/app.js#L42

**Comment by aguarino77** [comment]
I agree that your method works globally.
But the example you cite is not appropriate, because `scales` is passed in the `options` object of the Chart line. `fill` must be passed in the **`data`** structure, like `label`. See http://www.chartjs.org/docs/#line-chart-data-structure. That's why I think dataset should be built by extending not only label, data and colors, but also more options as I suggest above. The same applies for dataset options like `lineTension`

**Comment by jtblin** [comment]
Well I see your point, but it is not correct for `fill: false`, this is the result of using the option I pointed above on the tick chart in the [main example](https://github.com/jtblin/angular-chart.js/blob/chartjs-2.0/examples/app.js#L181):

```
$scope.options = {
(..)
      elements: {
        line: {
          borderWidth: 0.5,
          fill: false
        },
        point: {
          radius: 0
        }
      },
(...)
    };
```

![image](https://cloud.githubusercontent.com/assets/1388974/14941377/de85d9e4-0fdb-11e6-8a1a-0958ebcd6987.png)

vs without this option:

![image](https://cloud.githubusercontent.com/assets/1388974/14941383/1f38dcfc-0fdc-11e6-8c21-c14c66da9d1e.png)

**Comment by aguarino77** [comment]
Ok, now I see what you mean. Referring to [https://github.com/chartjs/Chart.js/blob/v2.0-dev/dist/Chart.js#L2626](url) you are using the second way to pass the options: they are not bound to the dataset but to the `elements` of the chart options.

It might be useful to mention this in the docs, once Chartjs 2 is officially out.

Thank you!

**Comment by jtblin** [comment]
Yes, the only limitation is that it's per chart, not per series. I will have to add the possibility to pass the dataset directly as pointed out in #131for that.

**Comment by courchef** [comment]
Hi guys,

[I wrote back in February](https://github.com/jtblin/angular-chart.js/issues/309#issuecomment-186468124) that I set `fill: false` in global options of Chart.js, but I realized afterward it wasn't suiting my needs since I have three charts on the same page. I ended up creating two option objects in my scope, one for the main chart and one for the two secondary charts, exatcly like Jerome suggest [in his previous comment](https://github.com/jtblin/angular-chart.js/issues/370#issuecomment-216033660).

You can view the result where the main chart, on the left, is drawn without filling. The main chart is a custom line chart that includes colored regions.
![chart js](https://cloud.githubusercontent.com/assets/15271147/14942381/56252296-0f89-11e6-83f0-b3e251ae8cee.png)

Thanks again Jerome for your work!

**Comment by jtblin** [comment]
It is now possible to override individual datasets options with #418. Published `1.0.0-alpha8` with this change.
